server/src/main/java/org/elasticsearch/indices/IndicesModule.java

```diff
@@ -157,6 +158,7 @@ public IndicesModule(List<MapperPlugin> mapperPlugins) {
         builtInMetadataMappers.put(IndexFieldMapper.NAME, new IndexFieldMapper.TypeParser());
         builtInMetadataMappers.put(SourceFieldMapper.NAME, new SourceFieldMapper.TypeParser());
         builtInMetadataMappers.put(TypeFieldMapper.NAME, new TypeFieldMapper.TypeParser());
```

Comment:
Can you switch to the `NestedPathFieldMapper.TypeParser` in this pr since you handle bwc in the new field type ?

Reply:
We may be able to in future, but I think we need to keep TypeFieldMapper around for the moment because it's still used for type queries, etc.

Reply:
Ok I was confused because there are some logic to handle bwc in NestedPathFieldMapper.